src/main/java/bgu/spl/mics/application/passiveObjects/Diary.java

```diff
@@ -38,23 +63,33 @@ public void addReport(Report reportToAdd){
 	 * List of all the reports in the diary.
 	 * This method is called by the main method in order to generate the output.
 	 */
-	public void printToFile(String filename){
-		//TODO: Implement this
+	public void printToFile(String filename) throws IOException {
+		_lock.readLock().lock();
+		Gson gson = new Gson();
+		String reportsToPrint = gson.toJson(_diary);
+		PrintWriter writer = null;
+		Path file = Paths.get(filename);
+		Files.write(file, Collections.singleton(reportsToPrint), StandardCharsets.UTF_8);
+		_lock.readLock().unlock();
 	}
 
 	/**
 	 * Gets the total number of received missions (executed / aborted) be all the M-instances.
 	 * @return the total number of received missions (executed / aborted) be all the M-instances.
 	 */
 	public int getTotal(){
-		//TODO: Implement this
-		return 0;
+		return _total.get();
 	}
 
 	/**
 	 * Increments the total number of received missions by 1
 	 */
 	public void incrementTotal(){
-
+		int localTotal;
```

Comment:
call it oldTotal, to  stick with convention

Reply:
Done